src/annotator/plugin/pdf.js

```diff
@@ -50,6 +90,69 @@ export default class PDF extends Delegator {
     return this.pdfMetadata.getMetadata();
   }
 
+  /**
+   * Check whether the PDF has selectable text and show a warning if not.
+   */
+  async _checkForSelectableText() {
+    // Wait for PDF to load.
+    try {
+      await this.uri();
+    } catch (e) {
+      return;
+    }
+
+    try {
+      const hasText = await pdfAnchoring.documentHasText();
+      this._showNoSelectableTextWarning(!hasText);
+    } catch (err) {
+      /* istanbul ignore next */
+      console.warn('Unable to check for text in PDF:', err);
+    }
+  }
+
+  /**
+   * Set whether the warning about a PDF's suitability for use with Hypothesis
+   * is shown.
+   *
+   * @param {boolean} showWarning
+   */
+  _showNoSelectableTextWarning(showWarning) {
+    if (!showWarning) {
+      this._warningBanner?.remove();
+      this._warningBanner = null;
+      return;
+    }
+
+    const mainContainer = /** @type {HTMLElement} */ (document.querySelector(
+      '#mainContainer'
+    ));
+
+    const updateBannerHeight = () => {
+      /* istanbul ignore next */
+      if (!this._warningBanner) {
+        return;
+      }
+      const rect = this._warningBanner.getBoundingClientRect();
+      mainContainer.style.top = rect.height + 'px';
+      this._warningBanner.style.top = -rect.height + 'px';
+    };
+
+    this._warningBanner = document.createElement('div');
+    this._warningBanner.className =
+      'annotator-pdf-warning-banner annotator-pdf-warning-banner--notice';
+    mainContainer.appendChild(this._warningBanner);
+    render(<WarningBanner />, this._warningBanner);
+
+    // @ts-expect-error - TS is missing `ResizeObserver`
+    if (typeof ResizeObserver !== 'undefined') {
+      // Update the banner when the window is resized or the Hypothesis
+      // sidebar is opened.
+      // @ts-ignore
+      new ResizeObserver(updateBannerHeight).observe(this._warningBanner);
```

Comment:
`ResizeObserver` is supported in, it looks like, all of our target browsers save for Android's Firefox. Historically, we've used `MutationObserver` (broader/older support but more complex API) as a polyfill, see `src/sidebar/util/observe-element-size` — do we have a plan going forward?

Reply:
`ResizeObserver` was only recently introduced in Safari (13.1) which means it isn't available on older iOS devices. In older browsers the banner height will simply be static based on the window size when the client first loads. I think that's an acceptable failure mode.